flink-connector-opensearch/src/main/java/org/apache/flink/connector/opensearch/sink/OpensearchWriter.java

```diff
@@ -343,7 +348,11 @@ private void extractFailures(BulkRequest request, BulkResponse response) {
         if (chainedFailures == null) {
             return;
         }
-        throw new FlinkRuntimeException(chainedFailures);
+        if (failureHandler == null) {
```

Comment:
This `null` check is not needed anymore (the builder would supply the default handler), just `failureHandler.onFailure(chainedFailures);`